admin_view_permission/admin.py

```diff
@@ -88,13 +88,15 @@ def has_change_permission(self, request, obj=None, only_change=False):
         """
         change_permission = super(AdminViewPermissionBaseModelAdmin,
                                   self).has_change_permission(request, obj)
-        if only_change:
-            return change_permission
+        if change_permission or self.has_view_permission(request, obj):
+            return True
         else:
-            if change_permission or self.has_view_permission(request, obj):
-                return True
-            else:
-                return change_permission
+            return change_permission
+
+    def has_change_permission_only_change(self, request, obj=None):
```

Comment:
Could you add some tests for this function? Also, could you make this function private as it is used internally only? I think `has_change_permission_only` it would be better name. The `_change` I think it is redundant.